### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by David-Verde**
Add Local storage to the Input Form
Update all.js files.
Install ESlint on project

### CODE REVIEW

File: oops.min.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-unused-vars */
```
Comment (Ismailco) [inline oops.min.js]: - [ ] Please don't ignore linters, you can try to fix them instead

File: oops.min.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+/* eslint-disable no-var */
+/* eslint-disable max-len */
+
+// eslint-disable-next-line require-jsdoc
+function Oops(a) {
```
Comment (Ismailco) [inline oops.min.js]: - [ ] The form data for all inputs should be stored in a single object, and saved to a single local storage registry, please implement this feature so your code meets the requirements

File: oops.min.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      if ('type' in h) {
+        b.def(f, g, h.value, true);
+      }
+    }); b.def(f, 'selected', e.length, true); return f;
+  }, noError: function(e) {
+    if (e=='localStorage'||e=='sessionStorage') {
+      if (b.storageAvailable(e)) {
+        return true;
+      } else {
+        console.warn('oops.js warn: '+e+' is not available.'); return false;
```
Comment (Ismailco) [inline oops.min.js]: - [ ] For best practice please consider not committing  ```console.warn``` or ```console.log``` to your repo it makes the code looks messy.

File: oops.min.js (hunk 2 above)
Comment (Ismailco) [inline oops.min.js]: - [ ] Your code look's too messy, if you can please consider using a non minified code so the code reviewer can check your code more easily

File: README.md [hunk 4]
```diff
@@ -19,6 +19,7 @@ Google Dev Tools
```
Comment (Ismailco) [inline README.md:19]: - [ ] [Optional] Your README needs some modification to look more professional, kindly consider changing the name to a meaningful one

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Unyielding1,

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Github Flow
✔️ Work documented professionally
✔️ The form behave as expected

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: form.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+      e.code === 22 ||
+      e.code === 1014 ||
+      e.name === 'QuotaExceededError' ||
+      e.name === 'ERROR_QUOTA_REACHED') && (storage && storage.length !== 0);
+  }
+}
+
```
Comment (sja-thedude) [inline form.js]: - [ ] As suggested by the previous reviewer, Kindly make sure that the form data for all inputs should be stored in a single object, and saved to a single local storage registry.

```
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```

File: form.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  inputData.text = formtext.value;
+  storage.setItem('formData', JSON.stringify(inputData));
+});
+
+formtext.addEventListener('change', () => {
+  inputData.name = formName.value;
+  inputData.email = email.value;
+  inputData.text = formtext.value;
+  storage.setItem('formData', JSON.stringify(inputData));
+});
```
Comment (sja-thedude) [inline form.js]: - [ ] Kindly make sure that when the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data. 
( _When the user refreshes the browser the data in the form should be persistent and to make it so you can follow the code below_ )

```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Unyielding1, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**